delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/FilesRoutesSpec.scala

```diff
@@ -473,6 +550,17 @@ class FilesRoutesSpec
         response.header[Location].value.uri shouldEqual Uri("https://bbp.epfl.ch/nexus/web/org/project/resources/file1")
       }
     }
+
+    "fail when auth fails" in {
+      val mockRoutes = createRoutes(filesWithErrorInFileContents[ResourceRejection](ResourceRejection.BlankResourceId))
+      Get("/v1/files/org/proj/file1") ~> Accept(`*/*`) ~> mockRoutes ~> check {
+        contentType.value shouldEqual `application/ld+json`.value
+        response.asJson shouldEqual jsonContentOf(
+          "/files/errors/blank-id.json"
+        )
+        status shouldEqual StatusCodes.BadRequest
+      }
+    }
```

Comment:
I don't see an alternative to creating a "mock" files which I can inject the error into. I realise that the `Files` interface is really too big, but there's not a lot I can do about it in the scope of this ticket

I'm using `ResourceRejection.BlankResourceId` only because it's easy to create

Reply:
The bug is related to `FileResponse` and not files right ?
`FileResponse` is also used in the archives plugin for example ?
What about setting a route that would emit a successful/failing `FileResponse` ? It should be a lot easier ?

Reply:
I would prefer to test the actual functionality where the bug was occurring, rather than make another situation where the bug would occur. I think the problem here is that our current design makes mocking this really hard. I think if we don't like the outcome of this, the solution is to fix our design rather than work around it.

I suppose it's possible that the archive functionality would have a problem if authentication failed, but I don't think it would manifest the same way. We would need a different test to catch that.

I realise this boilerplate isn't ideal, but other than that I think this test makes the most sense. If the boilerplate is annoying we can hide it

Reply:
We have already said we want to refactor the way that files are handled. When that happens, it would probably be very easy to improve this test. I think a test with a mock makes sense in principle, the only problem is that the design of `Files`

I'm not sure what `FilesRoutesSpec` is supposed to be testing, but given that we have `FilesSpec` as a real integration test, it seems like `FileRoutesSpec` being a unit test that handles this sort of thing makes a lot of sense

Reply:
I am not arguing that `Files` would need to be refactored but more about the fact that this mock on `Files` does not bring value when what you want to test concerns `FileResponse` and how it is sent to the user.
Besides having a dedicated test class for this in the sdk can be interesting for the following reasons:
* The test would live in the same module as `FileResponse` in the sdk and not in a related plugin, it is always better to have it closer when possible
* It won't be mixed in the folklore of Files and its numerous tests
* Until the refactoring is done, we will have to deal with this mock everytime we modify `Files`

Reply:
Those are good points, I think your original suggestion of a new routes returning a `FileResponse` makes the most sense. This would essentially be a unit test of `ResponseToJsonLd` (albeit only for the file response portion), which is basically the closest it can be